packages/material-ui-lab/src/DateRangePicker/makeDateRangePicker.tsx

```diff
@@ -65,23 +65,29 @@ export function makeDateRangePicker<TWrapper extends SomeWrapper>(
     areValuesEqual: (utils, a, b) => utils.isEqual(a[0], b[0]) && utils.isEqual(a[1], b[1]),
   };
 
-  function RangePickerWithStateAndWrapper<TDate>({
-    calendars,
-    value,
-    onChange,
-    mask = '__/__/____',
-    startText = 'Start',
-    endText = 'End',
-    inputFormat: passedInputFormat,
-    minDate: __minDate = defaultMinDate as TDate,
-    maxDate: __maxDate = defaultMaxDate as TDate,
-    ...other
-  }: BaseDateRangePickerProps<TDate> &
-    AllSharedDateRangePickerProps<TDate> &
-    PublicWrapperProps<TWrapper>) {
+  function RangePickerWithStateAndWrapper<TDate>(
+    props: BaseDateRangePickerProps<TDate> &
+      AllSharedDateRangePickerProps<TDate> &
+      PublicWrapperProps<TWrapper>,
+  ) {
+    const propsWithDefault = useThemeProps({ props, name });
```

Comment:
We should use the shortest name for the one that should be used throughout the component. I propose we use `inputProps` for the function parameter and then `props` for the final object.

That way we don't constantly have to change the actual implementation (e.g. by renaming variables).

`inputProps` is already used in other places and recognized by some internal static analysis tools.

Reply:
Agree, we should use the existing naming convention:

https://github.com/mui-org/material-ui/blob/a4b5e5537a824bb8b7521ac1146863047e97644f/packages/material-ui/src/Button/Button.js#L324-L325

Reply:
Disregard https://github.com/mui-org/material-ui/pull/24309#discussion_r553879716 since there's no reason given. The name should be `inputProps`.

Reply:
> no reason given

The initial comment was using the consistency argument.

Regarding why @mnajdova used this naming convention, if I had to guess I would assume: 

1. to minimize the diff during the refactoring. 2. to keep the rendering logic idiomatic (to use "prop").

But best to ask her.

Reply:
I'll follow https://github.com/mui-org/material-ui/pull/24309#discussion_r553744501. Does everyone agree? I found more occurrences of `inputProps` than `inProps`.

Reply:
I can count

`inputProps` x1, e.g
https://github.com/mui-org/material-ui/blob/a4b5e5537a824bb8b7521ac1146863047e97644f/packages/material-ui/src/Badge/Badge.js#L209-L210

`inProps`: x6, e.g.

https://github.com/mui-org/material-ui/blob/a4b5e5537a824bb8b7521ac1146863047e97644f/packages/material-ui/src/Button/Button.js#L324-L325

Reply:
>We will need to normalize to a single name.

Why?

Reply:
@eps1lon I don't have a preference in the final name we pick. What I care about is that we can resolve this question once for all. This is why I think normalizing to a single name is important. Once it's the case, you can follow hat the majority does when you don't know how things should be done, you can copy.

Reply:
> I think that we should resolve this question once for all. This is why I think normalizing to a single name is important (and no, not important enough to have a rule in the CI)

There's no reason given from your part. You just applied circular logic: "I think we should resolve it, therefore we should change it".

Reply:
@eps1lon Ok, I can be clearer: the same solution should be applied to the same problems. This is an axiom. I think we can all agree on it. Then comes to this specific case. We should make a case for why different names should be used, not the opposite.

Reply:
I don't care. Just don't break existing code and use `inputProps` where the name is required, thanks.

Reply:
I updated this PR to reflect the changes you all requested. In the end I ended up using `inProps` because `inputProps` was already being used. I don't think this PR is the best place to discuss these things. If consistency is a big concern let's open a RFC to discuss this and other inconsistencies. It would be nice to enrich the [API Design page](https://material-ui.com/guides/api/) with variable naming conventions.